cookbooks/retention.md

```diff
@@ -0,0 +1,211 @@
+*Authored by the Product Data Science Team. Please direct questions/concerns to Ben Miroglio (`bmiroglio`).*
+
+# Retention
+
+Retention measures the rate at which users are *continuing* to use Firefox, making it one of the more important metrics we track. We commonly measure retention between releases, experiment cohorts, and various Firefox subpopulations to better understand how a change to the user experience or use of a specific feature affect behavior.
+
+### N Week Retention
+
+Time is an embedded component of retention. Most retention analysis starts with some anchor, or action that is associated with a date (experiment enrollment date, profile creation date, button clicked on date *d*, etc.). We then look 1, 2, …, N weeks beyond the anchor to see what percent of users have submitted a ping (signaling their continued use of Firefox).
+
+For example, let’s say we are calculating retention for new Firefox users. Each user can then be anchored by their `profile_creation_date`, and we can count the number of users who submitted a ping between 7-13 days after profile creation (1 Week retention), 14-20 days after profile creation (2 Week Retention), etc.
+
+#### Example Methodology
```

Comment:
Would it be too difficult to add a pure SQL example?

Reply:
In addition to or in place of the current example?

Reply:
I think in addition to would be fine, the more examples the merrier (IMO).

Reply:
Agreed :). I'd add one in.

Reply:
I'm working on a solution that uses views, as i don't think it is realistic to do everything in one giant query. More for the folks who are more SQL-inclined and not super familiar with Pyspark.